src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/UseCrossPlatformIntrinsics.Fixer.cs

```diff
@@ -0,0 +1,140 @@
+                if (!Enum.TryParse(ruleKindName, out RuleKind ruleKind))
+                {
+                    continue;
+                }
+
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        UseCrossPlatformIntrinsicsTitle,
+                        c => ReplaceWithCrossPlatformIntrinsicAsync(context.Document, ruleKind, invocation, c),
+                        equivalenceKey: nameof(UseCrossPlatformIntrinsicsFixer)
+                    ),
+                    diagnostic
+                );
+            }
+        }
+
+        private async Task<Document> ReplaceWithCrossPlatformIntrinsicAsync(Document document, RuleKind ruleKind, IInvocationOperation invocation, CancellationToken cancellationToken)
+        {
+            DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            SyntaxGenerator generator = editor.Generator;
+
+            SyntaxNode? replacementNode = ruleKind switch
+            {
+                RuleKind.opAddition => ReplaceBinaryOp(invocation, isCommutative: true, generator.AddExpression),
+                RuleKind.opBitwiseAnd => ReplaceBinaryOp(invocation, isCommutative: true, generator.BitwiseAndExpression),
+                RuleKind.opBitwiseOr => ReplaceBinaryOp(invocation, isCommutative: true, generator.BitwiseOrExpression),
+                RuleKind.opDivision => ReplaceBinaryOp(invocation, isCommutative: false, generator.DivideExpression),
+                RuleKind.opExclusiveOr => ReplaceBinaryOp(invocation, isCommutative: true, CreateExclusiveOrExpression),
+                RuleKind.opLeftShift => ReplaceBinaryOp(invocation, isCommutative: false, CreateLeftShiftExpression),
+                RuleKind.opMultiply => ReplaceBinaryOp(invocation, isCommutative: true, generator.MultiplyExpression),
+                RuleKind.opOnesComplement => ReplaceUnaryOp(invocation, generator.BitwiseNotExpression),
+                RuleKind.opRightShift => ReplaceBinaryOp(invocation, isCommutative: false, CreateRightShiftExpression),
+                RuleKind.opSubtraction => ReplaceBinaryOp(invocation, isCommutative: false, generator.SubtractExpression),
+                RuleKind.opUnaryNegation => ReplaceUnaryOp(invocation, generator.NegateExpression),
+                RuleKind.opUnsignedRightShift => ReplaceBinaryOp(invocation, isCommutative: false, CreateUnsignedRightShiftExpression),
+                _ => null,
+            };
+
+            if (replacementNode is not null)
+            {
+                editor.ReplaceNode(invocation.Syntax, replacementNode.WithTriviaFrom(invocation.Syntax));
+                document = document.WithSyntaxRoot(editor.GetChangedRoot());
+            }
+
+            return document;
+
+            static SyntaxNode? ReplaceUnaryOp(IInvocationOperation invocation, Func<SyntaxNode, SyntaxNode?> unaryOpFunc)
+            {
+                if (invocation.Arguments.Length != 1)
```

Comment:
nit (since we shouldn't be doing checking in the fixer anyways), but this is nicer as: 

```suggestion
return invocation.Arguments is [{ Value.Syntax: var syntax }] ? unaryOpFunc(syntax) : null;
```

Reply:
I'd disagree on that being nicer. I find that much harder to read and comprehend, I believe it is trying to do "too much" on a single line.

Code does not need to be as short as possible, it should be written in a way that helps developers read and understand the intent and flow of the logic.

Reply:
i'm fine with teh user of statements versus expressions.  my general preference those is patterns over query/retrieval pairs.

Reply:
> The reason I find the above much better is because it uses words and simple symbols, r

Words are fine.  But i find actual structure nicer to model what the datastructure actually is (both when created) and stored in memory.  in other words `[var single]` is the clear way to say "a collection with one element in it, and this is the element"

Similarly with the otehr code you had, being able to do `[var firstParameter, var secondParameter]` means there's no asking myself how do i get the data.  Or how multiple checks related to extracting values out.   It's simply stating that the method has two parameters, and these are them :)

Reply:
> Can you make that happen :D

We've talked about it before and unfortunately there isn't a good name anyones come up with for this purpose 😓